base_location_nuts/models/res_partner.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Python source code encoding : https://www.python.org/dev/peps/pep-0263/
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright :
+#        (c) 2015 Antiun Ingenieria, SL (Madrid, Spain, http://www.antiun.com)
+#                 Antonio Espinosa <antonioea@antiun.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+from openerp.tools.translate import _
+import collections
+
+
+def dict_recursive_update(d, u):
+    for k, v in u.iteritems():
+        if isinstance(v, collections.Mapping):
+            r = dict_recursive_update(d.get(k, {}), v)
+            d[k] = r
+        else:
+            d[k] = u[k]
+    return d
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    region = fields.Many2one(comodel_name='res.partner.nuts',
+                             string="Region")
+    substate = fields.Many2one(comodel_name='res.partner.nuts',
+                               string="Substate")
+    lbl_region = fields.Char(compute='_labels_get')
+    lbl_substate = fields.Char(compute='_labels_get')
+
+    @api.one
+    @api.depends('country_id')
```

Comment:
Maybe, you have to see this other PRs to understand this:

* https://github.com/OCA/l10n-spain/pull/150 : ```l10n_es_location_nuts``` addon setting labels for matching NUTS levels to spanish states defined by ```l10n_es_toponyms``` addon
* https://github.com/OCA/l10n-germany/pull/11 : ```l10n_de_location_nuts``` addon setting labels for matching NUTS levels to german states defined by ```l10n_de_country_states``` addon

This is not only country dependant, but also localization dependant, that's why this is not a country config parameter

@pedrobaeza, you'll will have diferent labels in different partners if country is different. In examples:

* Partner A: Germany country, labels 'Government region' and 'Region'
* Partner B: Spain country, labels 'Autonomous Community' and 'Region'